web/src/billing/upgrade.ts

```diff
@@ -98,14 +98,17 @@ export const initialize = (): void => {
                         `/billing/event_status?stripe_payment_intent_id=${response_data.stripe_payment_intent_id}`,
                     );
                 } else if (response_data.organization_upgrade_successful) {
-                    window.location.replace("/billing");
+                    helpers.redirect_to_billing_with_successful_upgrade();
                 }
             },
-            () => {
+            (xhr) => {
                 $("#org-upgrade-button-text").show();
                 $("#org-upgrade-button .upgrade-button-loader").hide();
-                const error_text = $error_box.text();
-                $error_box.text(`${error_text} Please fix this issue or use a different card.`);
+                // Add a generic help text for card errors.
+                if (xhr.responseJSON.error_description === "card error") {
```

Comment:
This has invalidated this comment in `class BillingError`:

https://github.com/zulip/zulip/blob/0ef78c793e8bda3772a6c2477bf997f39ce2841d/corporate/lib/stripe.py#L371-L372

Reply:
Opened #29810, thanks for pointing out!